### ISSUE TITLE
Problem depositing article when manually entering journal title instead of using SHERPA lookup

### ISSUE THREAD

**Issue by cecross1** [VTUL/vtechworks#466]
[MerolaMuOxidoBis.pdf](https://github.com/VTUL/vtechworks/files/804090/MerolaMuOxidoBis.pdf)

Last Friday, I attempted to deposit the attached OASF article into VTechWorks. After I'd finished the first two submission screens and clicked Next to upload the PDF, I received an error message (please see attached screenshot). I tried logging out of VTechWorks, then closing and reopening Chrome, my default browser. I also tried resuming the submission in Firefox. Neither of these attempts seemed to help, and I kept getting the same error message.

I made other submissions without a problem earlier last week. I also tried again today, and got the same error message. @alawvt is aware of the issue, also. Please let me know if you have any questions. 

Update: I tried uploading a different article this morning, and didn't have any trouble with the submission. I'm wondering if there's a problem with the attached article, but I can open it on my computer without a problem, so I'm not sure what to think. Just wanted to let you know.

**Comment by keithgee** [comment]
I was able to submit this article and it's at https://vtechworks.lib.vt.edu/handle/10919/75184.
The metadata needs to be corrected - @cecross1 can you fix the metadata up to our standards and also get the correct characters into the title and description?

@alawvt the only unusual thing I found in the log file today was this:
```
2017-02-27 10:10:00,828 ERROR org.dspace.core.PluginManager @ Name collision in named plugin, implementation class="org.dspace.content.authority.LCNameAuthority", name="#"
```

I don't know that it is related.

**Comment by keithgee** [comment]
FYI submitted with the latest version of Safari.

**Comment by cecross1** [comment]
Keith,

Thanks for your help with the submission. I'll take a look at the metadata.

C

**Comment by alawvt** [comment]
@keithgee, thanks for depositing this and checking the log. I am still getting the error in both Firefox and Safari. It seems to be related to entering a title in the Journal Title box and not using the Lookup button. If I use the Lookup button it seems to deposit ok. If I don't, I get the error. Would you mind trying it?  Thanks.

**Comment by alawvt** [comment]
Philip reported the same problem this (Tuesday) morning.

**Comment by keithgee** [comment]
Yes, @alawvt, thank you. I'm able to duplicate the issue on my local development environment, where the log files are less busy and where there's no need to clean up after test items.

When I enter the journal title manually and never use the lookup button, I get the error screen when I try to go to the next page. When I submitted the original failed item to production, I used the Lookup button to find the journal.

**Comment by keithgee** [comment]
I have pushed branch fix_manual_entry_of_journal_title with a quick fix for this issue, PR #467. 

The fix allows manual entry of the journal title, and with the fix, SHERPA lookup still works. I've not looked at this particular code before, so it may be one of many alternative fixes.

I've added the dspacecore label to this issue, because we may wish to research and test if this bug still exists in DSpace 6 and 7, and if so, submit a problem report if one is needed and then a fix.

I have many questions.

Should this fix be deployed to VTechWorks production soon because it fixes a partially broken submission process, or should we wait for the next ordinary release? If we wait, should we temporarily disable the SHERPA lookup on production?

Is it likely that this is a problem that has existed on VTechWorks and we've never noticed it before, or is it instead something that a recent change to VTechWorks may have caused?

**Comment by alawvt** [comment]
@keithgee, thank you very much for this fix. If you'd like to go back a ways in the commit history to see if this was a problem earlier, that would be great. 

I would like to push this soon. The last deployment to dev, pprd, and prod were before the commits on 2/14-15 for the journal look up behavior. I assume that your branch was after those commits. I will pull your branch and test. If they all work fine, I'd like to push this to prod_deploy, etc. as a hot fix. I tagged the commit at 2017_02_28_hotfix. Next time we should try to use the procedure in https://github.com/VTUL/vtechworks/wiki/08.-Fixing-urgent-problems-on-production-(hotfixes).

We have been using the dspacecore tag for work that we'd like to submit to the main DSpace project. I have added a task to report this bug to DSpace. Other tasks could be to test this problem on DSpace 6 and to offer the change to DSpace.

**Comment by keithgee** [comment]
Thanks, @alawvt. It sounds like I would have done better to base my fix on what is currently deployed to production instead of the latest vt_5_x_dev commit.

**Comment by alawvt** [comment]
The initial issue to implement the Journal Title field lookup is #174. 

The DSpace 6 demo site, https://demo.dspace.org/xmlui/, does not have the journal title as part of its submission form. I would assume the [SHERPARoMEOJournalTitle](plugin9https://wiki.duraspace.org/display/DSPACE/Authority+Control+of+Metadata+Values#AuthorityControlofMetadataValues-ChoiceManagementConfiguration) plug-in is not implemented, but I don't know that for a fact.

Issue [Upload page fails when dc.title.alternative is enabled in config/spring/api/sherpa.xml and Sherpa/Romeo comes across a journal title it doesn't know](https://jira.duraspace.org/browse/DS-2751?jql=text%20~%20%22sherpa%22) and related Pull Request, [fixed DS-2751 by checking for null #1400](https://github.com/DSpace/DSpace/pull/1400) seems very similar to this issue. (We had changed this bean to work on dc.identifier.issn rather than dc.title.alternative in commit 182d6b3.) The fix is also for a null value of eISSN. That pull request has been merged in dSpace-6_x, but not dspace-5_x. I am going to try that fix, as well and the one by @keithgee, since it may fix the null value problem in an earlier step and would be more like the main DSpace files.

**Comment by alawvt** [comment]
My attempt to implement a version of [fixed DS-2751 by checking for null #1400](https://github.com/DSpace/DSpace/pull/1400) failed to fix the issue. However PR #467 that @keithgee wrote does, so I have merged that pull request and ask ITS to deploy it as a hot fix on production tomorrow morning.

**Comment by alawvt** [comment]
@keithgee's PR has been installed as a hot fix on production.
